src/compiler/utilities.ts

```diff
@@ -2680,24 +2680,28 @@ namespace ts {
             isClassLike(node.parent.parent);
     }
 
-    // Returns false if this heritage clause element's expression contains something unsupported
-    // (i.e. not a name or dotted name).
-    export function isSupportedExpressionWithTypeArguments(node: ExpressionWithTypeArguments): boolean {
-        return isSupportedExpressionWithTypeArgumentsRest(node.expression);
+    export function isSupportedExpressionWithTypeArguments(node: ExpressionWithTypeArguments): node is SupportedExpressionWithTypeArguments {
+        return isEntityNameExpression(node.expression);
     }
 
-    function isSupportedExpressionWithTypeArgumentsRest(node: Expression): boolean {
-        if (node.kind === SyntaxKind.Identifier) {
-            return true;
-        }
-        else if (isPropertyAccessExpression(node)) {
-            return isSupportedExpressionWithTypeArgumentsRest(node.expression);
-        }
-        else {
-            return false;
+    export function isEntityNameExpression(node: Expression): node is EntityNameExpression {

```

Comment:
Again, a better question for @vladima since he knows more about performance.

---
In reply to: [72693482](https://github.com/Microsoft/TypeScript/pull/10013#discussion_r72693482) [](ancestors = 72693482)